sinabs/backend/dynapcnn/chips/dynapcnn.py

```diff
@@ -174,64 +205,102 @@ def get_dynapcnn_layer_config_dict(cls, layer: DynapcnnLayer):
 
     @classmethod
     def write_dynapcnn_layer_config(
-        cls, layer: DynapcnnLayer, chip_layer: "CNNLayerConfig"
-    ):
+        cls,
+        layer: DynapcnnLayer,
+        layer2core_map: Dict[int, int],
+        destination_indices: List[int],
+        chip_layer: CNNLayerConfig,
+    ) -> None:
         """Write a single layer configuration to the dynapcnn conf object.
 
+        Uses the data in `layer` to configure a `CNNLayerConfig` to be
+        deployed on chip.
+
         Parameters
         ----------
-            layer:
-                The dynapcnn layer to write the configuration for
-            chip_layer: CNNLayerConfig
-                DYNAPCNN configuration object representing the layer to which
-                configuration is written.
+        - layer (DynapcnnLayer): Layer instance from which to generate the config
+        - layer2core_map (Dict): Keys are layer indices, values are corresponding
+            cores on hardware. Needed to map the destinations.]
+        - destination_indices (List): Indices of destination layers for `layer`
+        - chip_layer (CNNLayerConfig): Configuration object of the corrsesponding
+            on-chip core. Will be changed in-place based on `layer`.
         """
-        config_dict = cls.get_dynapcnn_layer_config_dict(layer=layer)
-        # Update configuration of the DYNAPCNN layer
-        chip_layer.dimensions = config_dict["dimensions"]
-        config_dict.pop("dimensions")
-        for i in range(len(config_dict["destinations"])):
-            if "pooling" in config_dict["destinations"][i]:
-                chip_layer.destinations[i].pooling = config_dict["destinations"][i][
-                    "pooling"
-                ]
-        config_dict.pop("destinations")
+
+        # extracting from a DynapcnnLayer the config. variables for its CNNLayerConfig.
+        config_dict = cls.get_dynapcnn_layer_config_dict(
+            layer=layer,
+            layer2core_map=layer2core_map,
+            destination_indices=destination_indices,
+        )
+
+        # update configuration of the DYNAPCNN layer.
+        chip_layer.dimensions = config_dict.pop("dimensions")
+
+        # set the destinations configuration.
+        for dest_idx, destination in enumerate(config_dict.pop("destinations")):
+            chip_layer.destinations[dest_idx].layer = destination["layer"]
+            chip_layer.destinations[dest_idx].enable = destination["enable"]
+            chip_layer.destinations[dest_idx].pooling = destination["pooling"]
+
+        # set remaining configuration.
         for param, value in config_dict.items():
             try:
                 setattr(chip_layer, param, value)
             except TypeError as e:
                 raise TypeError(f"Unexpected parameter {param} or value. {e}")
 
     @classmethod
-    def build_config(cls, model: "DynapcnnNetwork", chip_layers: List[int]):
-        layers = model.sequence
+    def build_config(
```

Comment:
Docstring for arguments needs to be updated (some args missing and some of the ones currently described do not exist).

Reply:
Can we also keep the parameters in the same order as much as possible?
Most methods I noticed use (Layers, layer2core_map, destination).